substrate/client/network/sync/src/justification_requests.rs

```diff
@@ -130,13 +187,21 @@ impl<B: BlockT> ExtraRequests<B> {
 		// currently enforced by the outer network protocol before passing on
 		// messages to chain sync.
 		if let Some(request) = self.active_requests.remove(&who) {
+			if let Some(metrics) = &self.metrics {
+				metrics.active.dec();
+			}
```

Comment:
Would grouping the request hashmaps (or vec) with metrics help here?
Some time in the future we might do a `active_requests.remove()` and forget to decrement the appropriate metric.

One suggestion might be to group them into a wrapper:
```rust
struct ActiveRequestsMetered {
    inner: HashMap<..>, // Current self.active_requests
    metrics: Option<Metrics>,
}
```

Maybe an even simpler approach would be to introduce a `fn active_requests_remove() { ...; metrics.active.dec(); }`.
Could also be a followup if would take too long to implement :D

Reply:
There are several of these in the code and I'm not sure how much it would help with ergonomics to be completely honest. Metrics are one of those things that are just inherently invasive unfortunately.